src/doc/rustc/src/targets/index.md

```diff
@@ -11,3 +11,5 @@ To compile to a particular target, use the `--target` flag:
 ```bash
 $ rustc src/main.rs --target=wasm32-unknown-unknown
 ```
+
+Please note, that the `--target` feature is generally considered as unsafe. More details can be found in [this section](https://doc.rust-lang.org/rustc/targets/known-issues.html).
```

Comment:
This may be a misunderstanding. The `--target` flag is fine AFAIK. I believe we simply never link artifacts compiled for  different targets together. The referenced issue is about the `-C target-feature` flag, which modifies what code is generated. A different `--target` can affect the same code generation decisions but AFAIK can't lead to ABI mismatches due to linking differently-compiled crates together.